docker/Dockerfile

```diff
@@ -0,0 +1,94 @@
+# Base image for the container - using Eclipse Temurin with OpenJDK 17 and Jammy base
+# This is the same image used in the offical Spark dockerfile which was used as reference
+# for this dockerfile
+FROM eclipse-temurin:17-jdk-jammy
+
+# Set environment variable to avoid interactive prompts during package installation
+# especially with krb5-user
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Update package lists and install required packages
+# The most important are bash gosu libnss-wrapper git ca-certificates
+# The rest might be removed in the future.
+# hadolint suggests adding `--no-install-recommends` to avoid additional packages so
+# it should be added in later versions.
+RUN set -ex; \
+    apt-get update; \
+    apt-get install -y gnupg2 wget bash tini libc6 libpam-modules krb5-user libnss3 procps net-tools gosu libnss-wrapper git ca-certificates; \
+    # Remove temporary files to reduce image size
```

Comment:
I want to hear your ideas about multi-stage docker builds. To reduce the size more effectively, another option would be to build a multi-stage Docker image. The first stage with the JDK produces a fat-jar, that can be copied to a JRE jar.

Reply:
I'll be completely honest here, I never used multi-stage with docker but I can see how it benefits projects that can be made of multiple modules. For instance with Spark, you can choose a flavor of Spark with Scala, with Scala and Python, with Scala and R etc...
In this case, it makes sense to have stages to build the various components.

In the case of unity catalog, I don't see how we can benefit from it.

From a size reduction perspective, we can certainly do a lot more to keep the size as small as possible, and among the things we can do is what you described as one fat-Jar.

We can also avoid optional dependencies in apt-get and choose an image smaller than the current `eclipse-temurin:17-jdk-jammy` such as `eclipse-temurin:22-jdk-alpine`.

My suggestion is to start somewhere at least with something, and iteratively improve and refine.

What do you think?

Reply:
I fully agree, it is more a nice to have than a must have!

Reply:
@Fokko Hello,
quick update on the multi-stage saga :laughing: 
I think I've got something that you will like with multi-stage, and very similar to what you described.
Stage 1 - compile a fat jar
stage 2 - copy the jar into a clean instance and run it

I'm experimenting with it since yesterday, I'll let you know as soon as I come up with something clean enough to be pushed into the branch without getting myself trolled. :rofl:

Reply:
I won't troll you, I promise! I'm also fine doing that in a follow-up PR to keep this one nice and small.